sqlmesh/core/config/connection.py

```diff
@@ -71,7 +72,7 @@ def _cursor_init(self) -> t.Optional[t.Callable[[t.Any], None]]:
         """A function that is called to initialize the cursor"""
         return None
 
-    def create_engine_adapter(self) -> EngineAdapter:
+    def create_engine_adapter(self, register_comments: bool = False) -> EngineAdapter:
```

Comment:
Want to call this `register_comments_override`? At first I was confused why this was needed since you define if you want to register comments at the connection config level but I think you have this so test connections can override. Let me know if I am missing something.